sample-api/clients/dotnet/src/Program.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Kiota.Authentication.Azure;
+using Microsoft.Kiota.Http.HttpClientLibrary;
+using ToDoClient.ApiClient;
+using ToDoClient.ApiClient.Models.ToDoApi.Models;
+
+var settings = Settings.LoadSettings();
+if (!settings.Validate())
+{
+    Console.WriteLine("Missing or invalid configuration values in appsettings.json.");
+    Console.WriteLine("See README.md for more information.");
+    return;
+}
+
+var scopes = new[] { settings.ApiScope ?? "" };
+var credential = new DeviceCodeCredential((code, cancellation) =>
+{
+    Console.WriteLine(code.Message);
+    return Task.FromResult(0);
+},
+settings.TenantId,
+settings.ClientId);
+
+var context = new TokenRequestContext(scopes);
+var response = await credential.GetTokenAsync(context);
+
+Console.WriteLine($"Token: {response.Token}");
+
+var authProvider = new AzureIdentityAuthenticationProvider(credential, scopes);
```

Comment:
once https://github.com/microsoft/kiota/pull/1051 gets merged, you might want to pass in the allowed hosts so it keeps working.